kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/MctsBeneficiaryUpdateServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.supercsv.cellprocessor.Optional;
+import org.supercsv.cellprocessor.ift.CellProcessor;
+import org.supercsv.prefs.CsvPreference;
+
+import javax.validation.ConstraintViolationException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implementation of the {@link MctsBeneficiaryUpdateService} interface.
+ */
+@Service("mctsBeneficiaryUpdateService")
+public class MctsBeneficiaryUpdateServiceImpl implements MctsBeneficiaryUpdateService {
+
+    private SubscriberService subscriberService;
+    private SubscriptionErrorDataService subscriptionErrorDataService;
+    private LocationService locationService;
+    private MctsMotherDataService mctsMotherDataService;
+    private MctsChildDataService mctsChildDataService;
+
+    private static final String SR_NO = "Sr No";
+    private static final String MCTS_ID = "MCTS ID";
+    private static final String STATE_MCTS_ID = "STATE ID";
+    private static final String DOB = "Beneficiary New DOB change";
+    private static final String LMP = "Beneficiary New LMP change";
+    private static final String MSISDN = "Beneficiary New Mobile no change";
+    private static final String STATE = "StateID";
+    private static final String DISTRICT = "District_ID";
+    private static final String TALUKA = "Taluka_ID";
+    private static final String HEALTH_BLOCK = "HealthBlock_ID";
+    private static final String PHC = "PHC_ID";
+    private static final String SUBCENTRE = "SubCentre_ID";
+    private static final String CENSUS_VILLAGE = "Village_ID";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MctsBeneficiaryUpdateServiceImpl.class);
+
+    @Autowired
+    public MctsBeneficiaryUpdateServiceImpl(SubscriberService subscriberService,
+                                            SubscriptionErrorDataService subscriptionErrorDataService,
+                                            LocationService locationService, MctsMotherDataService mctsMotherDataService,
+                                            MctsChildDataService mctsChildDataService) {
+        this.subscriberService = subscriberService;
+        this.subscriptionErrorDataService = subscriptionErrorDataService;
+        this.locationService = locationService;
+        this.mctsMotherDataService = mctsMotherDataService;
+        this.mctsChildDataService = mctsChildDataService;
+    }
+
+
+    @Override
+    public void updateBeneficiaryData(Reader reader) throws IOException {
+        BufferedReader bufferedReader = new BufferedReader(reader);
+
+        CsvMapImporter csvImporter = new CsvImporterBuilder()
+                .setProcessorMapping(getProcessorMapping())
+                .setPreferences(CsvPreference.STANDARD_PREFERENCE)
+                .createAndOpen(bufferedReader);
+        try {
```

Comment:
In general, we shouldn't. There are a few places (here as well as in the FLW import code) that this should be cleaned up.